api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchasesAPI.kt

```diff
@@ -127,45 +148,236 @@ private class PurchasesAPI {
+            storeProduct,
+            upgradeInfo
+        ) { _: StoreTransaction?, _: CustomerInfo -> }
+        purchases.purchaseProductWith(
+            activity,
+            storeProduct,
+            upgradeInfo,
+            true
+        ) { _: StoreTransaction?, _: CustomerInfo -> }
+        // endregion
+        // purchasePackageWith product change listenerConversions
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            upgradeInfo,
+            onError = { _: PurchasesError, _: Boolean -> },
+            onSuccess = { _: StoreTransaction?, _: CustomerInfo -> }
+        )
         purchases.purchasePackageWith(
             activity,
             packageToPurchase,
             upgradeInfo,
+            true,
             onError = { _: PurchasesError, _: Boolean -> },
             onSuccess = { _: StoreTransaction?, _: CustomerInfo -> }
         )
         purchases.purchasePackageWith(
             activity,
             packageToPurchase,
+            upgradeInfo,
+            onSuccess = { _: StoreTransaction?, _: CustomerInfo -> }
+        )
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            upgradeInfo,
+            true,
+            onSuccess = { _: StoreTransaction?, _: CustomerInfo -> }
+        )
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            upgradeInfo
+        ) { _: StoreTransaction?, _: CustomerInfo -> }
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            upgradeInfo,
+            true
+        ) { _: StoreTransaction?, _: CustomerInfo -> }
+        // endregion
+        // purchasePackageWith listenerConversions
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            onError = { _: PurchasesError, _: Boolean -> },
+            onSuccess = { _: StoreTransaction, _: CustomerInfo -> }
+        )
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            true,
+            onError = { _: PurchasesError, _: Boolean -> },
+            onSuccess = { _: StoreTransaction, _: CustomerInfo -> }
+        )
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            onSuccess = { _: StoreTransaction, _: CustomerInfo -> }
+        )
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            true,
+            onSuccess = { _: StoreTransaction, _: CustomerInfo -> }
+        )
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase
+        ) { _: StoreTransaction, _: CustomerInfo -> }
+        purchases.purchasePackageWith(
+            activity,
+            packageToPurchase,
+            true
+        ) { _: StoreTransaction, _: CustomerInfo -> }
+        // endregion
```

Comment:
this method got really long... i added regions, but maybe we should split out into mroe functions

Reply:
I agree, splitting into different functions would be more readable 👍